internal/utils/aggregator.go

```diff
@@ -33,9 +28,15 @@ func Aggregate_data(df dataframe.DataFrame) (dataframe.DataFrame, error) {
 	}
 
 	if df.Nrow() == 0 {
+		log.Error("No valid records present in CSV to process further.")
 		return df, nil
```

Comment:
I think here we should return the err instead of nil? because in dependent func we are checking if there is nil, if not nil then we end up processing the df further?

https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/services/report_processor.go#L75

Reply:
Yes, you are right. 
Thanks. I will make the changes.